### ISSUE TITLE
[BUG] selection with negative slices fails in dataframes with StringColumns

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#1489]
**Describe the bug**
When trying to use negative slicing to get rows of a dataframe containing StringColumns, we fail in at least two ways (depending on the ordering of columns). This affects direct index access, `iloc`, and `tail`.

If the first column in the dataframe is a StringColumn, we can't actually do the slicing at all.

If any other column is a StringColumn, the "preceding" columns are returned as empty in the output view. 

**Steps/Code to reproduce bug**

First column StringColumn
```
import cudf

df = cudf.DataFrame({'a':['a','b','c','d','e'],
                     'b':range(5),
                     'c':range(5),
                     'd':range(5)
                    })

print(df.iloc[-5:])
print(df.tail()) # same error
print(df[-5:]) # same error
    print(df[-5:])
  File "/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.0.dev0+791.g9bc62935.dirty-py3.7-linux-x86_64.egg/cudf/dataframe/dataframe.py", line 433, in __str__
    return self.to_string(nrows=nrows, ncols=ncols)
  File "/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.0.dev0+791.g9bc62935.dirty-py3.7-linux-x86_64.egg/cudf/dataframe/dataframe.py", line 422, in to_string
    cols[h] = self[h].values_to_string(nrows=nrows)
  File "/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.0.dev0+791.g9bc62935.dirty-py3.7-linux-x86_64.egg/cudf/dataframe/series.py", line 334, in values_to_string
    out = [str(v) for v in values]
  File "/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.0.dev0+791.g9bc62935.dirty-py3.7-linux-x86_64.egg/cudf/dataframe/series.py", line 334, in <listcomp>
    out = [str(v) for v in values]
  File "/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.0.dev0+791.g9bc62935.dirty-py3.7-linux-x86_64.egg/cudf/dataframe/series.py", line 286, in __getitem__
    return self._column.element_indexing(arg)
  File "/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.0.dev0+791.g9bc62935.dirty-py3.7-linux-x86_64.egg/cudf/dataframe/column.py", line 423, in element_indexing
    val = self.data[index]  # this can raise IndexError
  File "/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.0.dev0+791.g9bc62935.dirty-py3.7-linux-x86_64.egg/cudf/dataframe/buffer.py", line 144, in __getitem__
    return item.view(self.dtype)
AttributeError: 'NoneType' object has no attribute 'view'
```

Non-first column stringColumn

```
import cudf

df = cudf.DataFrame({'a':range(5),
                     'b':range(5),
                     'e':['a','b','c','d','e'],
                     'c':range(5),
                     'd':range(5)
                    })

print(df[-5:])
   a  b  e  c  d
0           0  0
1           1  1
2           2  2
3           3  3
4           4  4
```

**Expected behavior**
I expect negative slicing to succeed with StringColumns.

**Comment by beckernick** [comment]
I believe the issue is that nvstrings objects do not support negative slicing.

```
import nvstrings

s = nvstrings.to_device(['a', 'b', 'c', 'd'])
print(s[:3])
print(s[-3:])
['a', 'b', 'c']
[]
```

Going to mark this as a libcudf issue here, but I think it's actually a cuStrings issue and will open this on cuStrings. @kkraus14

**Comment by beckernick** [comment]
For tracking purposes: https://github.com/rapidsai/custrings/issues/257

**Comment by shwina** [comment]
Looks like this can be closed. Can you confirm @beckernick?

**Comment by beckernick** [comment]
Thanks @shwina . Yes, this has been resolved by rapidsai/custrings#268. Closing.
